components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/ServerApplicationManagementService.java

```diff
@@ -174,41 +181,72 @@ public ApplicationListResponse getAllApplications(Integer limit, Integer offset,
-                isEqualFilterUsed = true;
+        if (StringUtils.isNotBlank(filter)) {
+            try {
+                FilterTreeBuilder filterTreeBuilder = new FilterTreeBuilder(filter);
+                Node rootNode = filterTreeBuilder.buildTree();
+                if (rootNode instanceof ExpressionNode) {
+                    ExpressionNode expressionNode = (ExpressionNode) rootNode;
+                    if (!SEARCH_SUPPORTED_ATTRIBUTES.contains(expressionNode.getAttributeValue())) {
+                        throw buildClientError(ErrorMessage.UNSUPPORTED_FILTER_ATTRIBUTE, expressionNode
+                                .getAttributeValue());
+                    }
+                    formattedFilter = generateFilterStringForBackend(expressionNode.getAttributeValue(),
+                                expressionNode.getOperation(), expressionNode.getValue());
+
+                } else if (rootNode instanceof OperationNode) {
+                    // Currently, support only filters with one operation.
+                    // Have to recursively traverse the filter tree to support more than one operation.
+                    OperationNode operationNode = (OperationNode) rootNode;
+                    Node leftNode = rootNode.getLeftNode();
+                    Node rightNode = rootNode.getRightNode();
+
+                    if (leftNode instanceof ExpressionNode && rightNode instanceof ExpressionNode) {
+                        ExpressionNode expressionLeftNode = (ExpressionNode) leftNode;
+                        ExpressionNode expressionRightNode = (ExpressionNode) rightNode;
+                        if (!SEARCH_SUPPORTED_ATTRIBUTES.contains(expressionLeftNode.getAttributeValue())) {
+                            throw buildClientError(ErrorMessage.UNSUPPORTED_FILTER_ATTRIBUTE, expressionLeftNode
+                                    .getAttributeValue());
+                        }
+                        if (!SEARCH_SUPPORTED_ATTRIBUTES.contains(expressionRightNode.getAttributeValue())) {
+                            throw buildClientError(ErrorMessage.UNSUPPORTED_FILTER_ATTRIBUTE, expressionRightNode
+                                    .getAttributeValue());
+                        }
+
+                        String formattedLeftNode = generateFilterStringForBackend(
+                                expressionLeftNode.getAttributeValue(),
+                                expressionLeftNode.getOperation(),
+                                expressionLeftNode.getValue());
+                        String formattedRightNode = generateFilterStringForBackend(
+                                expressionRightNode.getAttributeValue(),
+                                expressionRightNode.getOperation(),
+                                expressionRightNode.getValue());
+                        formattedFilter = formattedLeftNode + " " +
+                                operationNode.getOperation() + " " + formattedRightNode;
```

Comment:
What are the supported `operationNode.getOperation()` operations ? Do we support both AND / OR?
Don't we need to validate the value of the operation of the operational node

Reply:
Also need to test out this filtering mechanism in every type of supported DBs (MySQL, h2, mssql, db2, postgres, oracle) in product-is